packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -429,7 +429,11 @@ export class TdsDropdown {
 
   private setValueAttribute = () => {
     if (this.value?.toString() === '') this.value = null;
-    this.host.setAttribute('value', this.value?.map((val) => val).toString() ?? null);
+    if (this.value === null) {
+      this.host.removeAttribute('value');
+    } else {
+      this.host.setAttribute('value', this.value?.map((val) => val).toString());
+    }
   };
```

Comment:
Should probably check if it null first, if not null the the guards "?" aren't needed.

Example: (haven't tried this, just psuedo code ish)
```jsx
  private setValueAttribute = () => {
    if(!value || this.value?.toString() === ''){
         this.host.removeAttribute('value');
         this.value = null;
    }  else {
      this.host.setAttribute('value', this.value.map((val) => val).toString());
      this.value = value;
    }
  };
```

Reply:
How come this `setValueAttribute ` method doesn't have any args? Is it used by us internally or by users?

Reply:
I’m a bit unsure about your comment regarding internal usage, as a private function is not publicly accessible. Additionally, regarding the arguments, the function was created 12 months ago without any parameters.

Reply:
I get that the `private` methods are exposed via the API to our users. But the `reset` method uses a `internatlReset` which then uses `setValueAttribute` so it is sort of exposed. But the `setValueAttribute` then just updates the attribute on the element. If that is the case I think the code I suggested above should be something more like:


```jsx
  private setValueAttribute = () => {
    if(!value || this.value?.toString() === ''){
         this.host.removeAttribute('value');
    }  else {
      this.host.setAttribute('value', this.value.map((val) => val).toString());
    }
  };
```

Since it should **only** update the value attribute and not property.

Reply:
If the value should also be "reset", I'd suggested either renaming this private method and remove the value here aswell. Or to create a setValue method (if it doesn't already exist) and call that after this method has been called, with a null or empty value.

Reply:
Thank you for the explanation @mJarsater , grateful for the input! 🙌